### PR TITLE
no weight decay in depthwise convolution

### PR DESCRIPTION
**Pull request by rydenisbak**
PR from this issue
https://github.com/open-mmlab/mmdetection/issues/2063

### CODE REVIEW

**Comment by rydenisbak** [comment]
I think you are right. Fixed

File: mmdet/apis/train.py [hunk 1]
```diff
@@ -162,6 +162,9 @@ def build_optimizer(model, optimizer_cfg):
         bias_lr_mult = paramwise_options.get('bias_lr_mult', 1.)
         bias_decay_mult = paramwise_options.get('bias_decay_mult', 1.)
         norm_decay_mult = paramwise_options.get('norm_decay_mult', 1.)
+        no_wd_in_dw = paramwise_options.get('no_wd_in_dw', False)
```
Comment (hellock) [inline mmdet/apis/train.py]: I suggest renaming this option to `dwconv_decay_mult` and use it as a multiplier.

File: mmdet/apis/train.py [hunk 2]
```diff
@@ -183,6 +186,14 @@ def build_optimizer(model, optimizer_cfg):
...
                 if base_wd is not None:
                     param_group['weight_decay'] = base_wd * bias_decay_mult
+            if no_wd_in_dw:
+                module_name = name.replace('.weight', '').replace('.bias', '')
+                if module_name in named_modules:
+                    module = named_modules[module_name]
+                    # if this Conv2d is depthwise Conv2d
+                    if isinstance(module, torch.nn.Conv2d) and \
+                            module.in_channels == module.groups:
+                        param_group['weight_decay'] = 0
```
Comment (hellock) [inline mmdet/apis/train.py]: Some optimizers do not accept the argument `weight_decay`. You may check if `base_wd` is not None and add an assertion in L159.

**Comment by hellock** [comment]
Thanks for your contribution! Please also update the docstring of `build_optimizer`.

**Comment by OpenMMLab-Assistant-007** [comment]
Hi!
@rydenisbak 
First of all, we want to express our gratitude for your significant PR in the OpenMMLab project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤
